### PR TITLE
chore: analytics improvements for reporting

### PR DESCRIPTION
**Pull request by saeedbashir**
This PR improves analytics for IAP events by sending LMS returned price, localized price, and localized currency code for better reporting as requested by the product.

### CODE REVIEW

**Review by rnr — APPROVED** [review]

File: Core/Core/CourseUpgrade/CourseUpgradeHelper.swift [hunk 1]
```diff
@@ -187,7 +199,7 @@ public class CourseUpgradeHelper: CourseUpgradeHelperProtocol {
         
         switch state {
         case .basket:
-            saveInProgressIAP(courseID: courseID, sku: sku)
+            saveInProgressIAP(courseID: courseID, sku: sku, lmsPrice: lmsPrice ?? 0.0)
```
Comment (shafqat-muneer) [inline Core/Core/CourseUpgrade/CourseUpgradeHelper.swift]: At few places, we are using default value as .zero. For consistency, please use same on all places.

File: Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift [hunk 2]
```diff
@@ -92,7 +95,9 @@ public class UpgradeInfoViewModel: ObservableObject {
                         courseID: self.courseID,
                         blockID: "",
                         pacing: pacing,
-                        coursePrice: "",
+                        localizedPrice: nil,
+                        localizedCurrencyCode: nil,
+                        lmsPrice: lmsPrice,
```
Comment (shafqat-muneer) [inline Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift:100]: These values accessible at this point. Shouldn't we use these values?
```
localizedPrice: product?.price,
localizedCurrencyCode: product?.currencySymbol,
```

File: Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift [hunk 3]
```diff
@@ -116,7 +121,9 @@ public class UpgradeInfoViewModel: ObservableObject {
                     courseID: self.courseID,
                     blockID: "",
                     pacing: pacing,
-                    coursePrice: nil,
+                    localizedPrice: nil,
```
Comment (shafqat-muneer) [inline Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift:124]: This values accessible at this point. Shouldn't we use these values?
```
localizedPrice: product?.price
```

File: OpenEdX/Data/DashboardPersistence.swift [hunk 4]
```diff
@@ -59,7 +59,8 @@ public class DashboardPersistence: DashboardPersistenceProtocol {
                                   progressPossible: 0,
                                   auditAccessExpires: $0.auditAccessExpires,
                                   startDisplay: $0.startDisplay,
-                                  startType: DisplayStartType(value: $0.startType))}
+                                  startType: DisplayStartType(value: $0.startType),
+                                  lmsPrice: $0.lmsPrice)}
```
Comment (shafqat-muneer) [inline OpenEdX/Data/DashboardPersistence.swift]: Please move closing brace of `.map` closure to next line.

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]

File: Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift (hunk 2 above)
Comment (saeedbashir) [reply Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift:100]: It's a price fetching error so in any case these will be nil.

File: Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift (hunk 3 above)
Comment (saeedbashir) [reply Core/Core/CourseUpgrade/ViewModels/UpgradeInfoViewModel.swift:124]: https://github.com/edx/edx-mobile-marketplace-ios/pull/16#discussion_r1671481443

File: Core/Core/CourseUpgrade/CourseUpgradeHelper.swift [hunk 5]
```diff
@@ -523,23 +539,27 @@ public class InProgressIAP: NSObject, NSCoding, NSSecureCoding {
...
         coder.encode(sku, forKey: "sku")
         coder.encode(pacing, forKey: "pacing")
+        coder.encode(lmsPrice, forKey: "lmsPrice")
     }
     
     public required init?(coder: NSCoder) {
         courseID = coder.decodeObject(forKey: "courseID") as? String ?? ""
         sku = coder.decodeObject(forKey: "sku") as? String ?? ""
         pacing = coder.decodeObject(forKey: "pacing") as? String ?? ""
+        lmsPrice = coder.decodeObject(forKey: "lmsPrice") as? Double ?? 0.0
```
Comment (shafqat-muneer) [inline Core/Core/CourseUpgrade/CourseUpgradeHelper.swift]: Please also update here `lmsPrice ?? 0.0` to `lmsPrice ?? .zero`

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]

**Review by shafqat-muneer — APPROVED** [review]
LGTM 👍

**Pull request merged**